packages/embark-specialconfigs/src/functionConfigs.js

```diff
@@ -76,8 +87,16 @@ class FunctionConfigs {
         // TODO: for this to work correctly we need to add a default from address to the contract
         if (contract.deploy === false) continue;
         // eslint-disable-next-line no-await-in-loop
-        let contractInstance = await this.events.request2("runcode:eval", contract.className);
-        args.contracts[contract.className] = contractInstance;
+        const contractRegisteredInVM = await this.checkContractRegisteredInVM(contract);
+        if (!contractRegisteredInVM) {
+          this.events.request("embarkjs:contract:runInVM", contract, async () => {
+            let contractInstance = await this.events.request2("runcode:eval", contract.className);
+            args.contracts[contract.className] = contractInstance;
+          });
+        } else {
+          let contractInstance = await this.events.request2("runcode:eval", contract.className);
+          args.contracts[contract.className] = contractInstance;
+        }
```

Comment:
You could simplify as so:
```
        if (!contractRegisteredInVM) {
          await this.events.request2("embarkjs:contract:runInVM", contract);
        }
        const contractInstance = await this.events.request2("runcode:eval", contract.className);
        args.contracts[contract.className] = contractInstance;
```

Reply:
Sweet! I went with a normal `request()` cause that's how it was done in another module that already has been refactored. Figured there might have been a reason it was using `request` instead of `request2`.

Will update!